src/communication/request_sync_client_server.hpp

```diff
@@ -42,46 +43,21 @@ namespace cubcomm
   {
     private:
 
-      // Wrap the user payload with a response sequence number
-      using response_sequence_number_t = std::uint64_t;
-      static constexpr response_sequence_number_t NO_RESPONSE =
-	      std::numeric_limits<response_sequence_number_t>::max ();
-
     public:
       using outgoing_msg_id_t = T_OUTGOING_MSG_ID;
       using request_client_server_t = cubcomm::request_client_server<T_OUTGOING_MSG_ID, T_INCOMING_MSG_ID>;
 
       // The user payload (of type T_PAYLOAD) is accompanied by a response sequence number set by
       // request_sync_client_server.
-      class sequenced_payload : public cubpacking::packable_object
-      {
-	public:
-	  sequenced_payload () = default;
-	  sequenced_payload (response_sequence_number_t a_rsn, T_PAYLOAD &&a_payload);
-	  sequenced_payload (sequenced_payload &&other);
-	  sequenced_payload (const sequenced_payload &other) = delete;
-	  ~sequenced_payload () = default;
-
-	  sequenced_payload &operator= (sequenced_payload &&other);
-	  sequenced_payload &operator= (const sequenced_payload &) = delete;
-
-	  void push_payload (T_PAYLOAD &&a_payload);
-	  T_PAYLOAD pull_payload ();
-
-	  size_t get_packed_size (cubpacking::packer &serializator, std::size_t start_offset = 0) const final override;
-	  void pack (cubpacking::packer &serializator) const final override;
-	  void unpack (cubpacking::unpacker &deserializator) final override;
-
-	private:
-	  response_sequence_number_t m_rsn = NO_RESPONSE;
-	  T_PAYLOAD m_user_payload;
-      };
+      class sequenced_payload;
 
       using incoming_request_handler_t = std::function<void (sequenced_payload &)>;
 
     public:
       request_sync_client_server (cubcomm::channel &&a_channel,
-				  std::map<T_INCOMING_MSG_ID, incoming_request_handler_t> &&a_incoming_request_handlers);
+				  std::map<T_INCOMING_MSG_ID, incoming_request_handler_t> &&a_incoming_request_handlers,
+				  T_OUTGOING_MSG_ID a_outgoing_response_msgid,
```

Comment:
outgoing and incoming response message ids could be template arguments, thus forcing unique declaration of classes (not that the latter is a problem)